web/static/js/components/respondents/RespondentsFilter.jsx

```diff
@@ -0,0 +1,68 @@
+// @flow
+import React, { Component } from 'react'
+import { translate } from 'react-i18next'
+import AwesomeDebouncePromise from 'awesome-debounce-promise'
+
+type Props = {
+  t: Function,
+  inputValue: string,
+  onChange: Function,
+  debounceMilliseconds: ?number,
+  onApplyFilter: Function
+}
+
+type State = {
+  inputValue: string,
+  applyFilterDebounced: Function
+}
+
+class RespondentsFilter extends Component<Props, State> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      inputValue: props.inputValue,
+      applyFilterDebounced: AwesomeDebouncePromise(
+        props.onApplyFilter,
+        props.debounceMilliseconds || 2000
```

Comment:
```suggestion
        props.debounceMilliseconds || 500
```

2 seconds is too much IMO.